opentreemap/api/tests.py

```diff
@@ -618,6 +636,48 @@ def test_update_plot(self):
 
         assert_reputation(self, reputation_count + 6)
 
+    def test_update_plot_with_udfs(self):
+        self._makeUdfd('abc', 'choice', choices=['a', 'b', 'c'])
+        self._makeUdfd('def', 'choice', choices=['d', 'e', 'f'])
+        test_plot = mkPlot(self.instance, self.user)
+        test_plot.width = 1
+        test_plot.length = 2
+        test_plot.geocoded_address = 'foo'
+        test_plot.udfs['abc'] = 'a'
+        test_plot.save_with_user(self.user)
+
+        self.assertEqual(test_plot.udfs['abc'], 'a')
+
+        # Include `updated_by` because the app does send it,
+        # and the endpoint must ignore it.
+        # Likewise, include `udf:def` for the same reason.
+        updated_values = {'plot':
+                          {'geom': {'y': 0.001, 'x': 0.001, 'srid': 4326},
+                           'width': 11,
+                           'length': 22,
+                           'udf:abc': 'b',
+                           'udf:def': None,
```

Comment:
I included this example data when I first reported the problem yesterday
https://github.com/OpenTreeMap/otm-core/pull/3124#discussion_r118564007